docs/architecture/adr/ADR043-k8s-resource-access.md

```diff
@@ -0,0 +1,530 @@
+daemonsets:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+deployments:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+ingresses:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+networkpolicies:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+podsecuritypolicies:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+replicasets:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+
+ApiGroup: metrics.k8s.io
+nodes:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+pods:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+
+ApiGroup: monitoring.coreos.com
+alertmanagers:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+podmonitors:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+prometheuses:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+prometheusrules:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+servicemonitors:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+
+ApiGroup: networking.istio.io
+destinationrules:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+envoyfilters:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+gateways:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+serviceentries:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+sidecars:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+virtualservices:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+
+ApiGroup: networking.k8s.io
+ingresses:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+networkpolicies:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+
+ApiGroup: node.k8s.io
+runtimeclasses:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+
+ApiGroup: policy
+poddisruptionbudgets:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+podsecuritypolicies:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+
+ApiGroup: queue.govsvc.uk
+sqs:
+  Operator:  create, delete, get, list, patch, update, watch
+  Auditor:   delete, get, list, watch
+
+ApiGroup: rbac.authorization.k8s.io
+clusterrolebindings:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+clusterroles:
+  Operator:  get, list, watch
+  Auditor:   get, list, watch
+rolebindings:
+  Operator:  create, delete, get, list, patch, update, watch
```

Comment:
I was worried about this, but I did some reading and it's fine. Leaving a comment in case anyone else is interested:

I was worried about this from a privilege escalation point of view.  For example, it seemed to me that if you can create arbitrary Roles and RoleBindings (not just ClusterRoles and ClusterRoleBindings), you can change which PodSecurityPolicy you use to be the most privileged one you can find that exists. Source: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#authorizing-policies

However, kubernetes has a control for this which prevents privilege escalation: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping In broad terms, a Role which can't already access a thing can't create a Role which does have access to a thing.  So in my particular example, a Role which doesn't already grant access to a PSP can't create a Role which does have access to a PSP.